core/pkg/monitor/network.go

```diff
@@ -49,6 +53,18 @@ func (n *Network) SampleMetrics() {
 			n.metrics["network.recv"],
 			float64(int(netIOCounters[0].BytesRecv)-n.recvInit),
 		)
+
+		n.metrics["network.upload_speed"] = append(
+			n.metrics["network.upload_speed"],
+			float64((int(netIOCounters[0].BytesSent)-n.lastSent)/2),
```

Comment:
You need to track when you measured BytesSent last time and what time it is now when you measure it again, that will be you delta_t, not 2.